### PR TITLE
Add AppVerifier

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
*Description of changes:*

Adds AppVerifier support to CI on Windows so when AppVerifier finds an issue it is surfaced easily. 

_______

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: utils/output.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<avrf:logfile xmlns:avrf="Application Verifier">
+	<avrf:logSession TimeStarted="2022-09-12 : 18:54:32" PID="16164" Version="2"/>
+</avrf:logfile>
```
Comment (TwistedTwigleg) [inline utils/output.xml:4]: This is a temporary file - will remove tomorrow morning 👍

File: utils/appverifier_ctest_util.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+# SPDX-License-Identifier: Apache-2.0.
+
+# Built-in
+from asyncio import subprocess
+import argparse
+import subprocess
+import os
+import json
+# Needs to be installed via pip
+from termcolor import colored  # - for terminal colored output
```
Comment (graebm) [inline utils/appverifier_ctest_util.py:11]: I personally prefer not installing dependencies over getting red output

just prefix errors with something easily searchable, like "ERROR:"

**Review by graebm — COMMENTED** [review]
let's chat in the morning about how to best pull this off

File: scripts/appverifier_xml.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    xml_is_app_verifier = False
+    app_verifier_entries = []
+
+    print("Looking for AppVerifier XML file...", flush=True)
+
+    try:
+        xml_tree = ElementTree.parse(filepath)
+    except Exception: # lgtm [py/catch-base-exception]
+        print("ERROR: Exception occured while trying to open XML file!", flush=True)
+        return -1
```
Comment (graebm) [inline scripts/appverifier_xml.py:171]: trivial: why bother trying to catch an exception only to kill the program? just let the exception happen

**Review by graebm — COMMENTED** [review]
lots of trivial python style comments, but looking good!

**Review by graebm — APPROVED** [review]

**Pull request merged**